File: src/main/java/com/github/rssanders3/spark/java/Main.java
package com.github.rssanders3.spark.java;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.SQLContext;
import org.apache.spark.sql.hive.HiveContext;
import org.kohsuke.args4j.CmdLineException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Arrays;
import java.util.List;

/**
 * Created by robertsanders on 11/9/16.
 */
public class Main {

    private static final Logger LOGGER = LoggerFactory.getLogger(MainArgs.class);

    private static String APP_NAME = "TO-BE-COMPLETED";

    public static void main(String[] args) throws CmdLineException {

        //if arguments include -help or --help then print Argument Usage
        List argsList = Arrays.asList(args);
        if(argsList.contains("-help") || argsList.contains("--help")) {
            new MainArgs().printUsage();
            System.exit(0);
        }

        MainArgs mainArgs = new MainArgs(args);
        LOGGER.info(mainArgs.toString());

        SparkConf conf = new SparkConf().setAppName(APP_NAME);
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new HiveContext(sc);


    }

}

File: src/main/scala/com/github/rssanders3/spark/scala/Main.scala
package com.github.rssanders3.spark.scala

import org.apache.spark.sql.hive.HiveContext
import org.apache.spark.{SparkConf, SparkContext}
import org.slf4j.{Logger, LoggerFactory}


/**
 * Created by robertsanders on 11/9/16.
 */
object Main {

  val LOGGER: Logger = LoggerFactory.getLogger(Main.getClass.getName)

  val APP_NAME: String = "TO-BE-COMPLETED"

  def main(args: Array[String]): Unit = {

    //if arguments include -help or --help then print Argument Usage
    if (args.contains("-help") || args.contains("--help")) {
      LOGGER.info(MainArgs.argsUsage)
      System.exit(0)
    }

    val jobArgs = MainArgs.parseJobArgs(args.toList)
    if (jobArgs == null) {
      LOGGER.error("")
      LOGGER.info(MainArgs.argsUsage)
      System.exit(-1)
    }

    jobArgs.validate()
    LOGGER.info(jobArgs.toString())

    val conf = new SparkConf().setAppName(APP_NAME)
    val sc = new SparkContext(conf)
    val sqlContext = new HiveContext(sc)


  }

}
